crates/bevy_core/src/time/time.rs

```diff
@@ -34,58 +49,171 @@ impl Time {
+        self.fixed_delta_seconds_f64 = self.fixed_delta.as_secs_f64();
+    }
+
+    /// The rate that time advances relative to the real world, as [`f32`].
+    #[inline]
+    pub fn relative_speed(&self) -> f32 {
+        self.relative_speed
+    }
+
+    /// The rate that time advances relative to the real world, as [`f64`].
+    #[inline]
+    pub fn relative_speed_f64(&self) -> f64 {
+        self.relative_speed as f64
+    }
+
+    /// Set rate time advances relative to the real world. Must be >= 0.0.
+    #[inline]
+    pub fn set_relative_speed(&mut self, relative_speed: f32) {
+        assert!(relative_speed >= 0.0);
+        self.relative_speed = relative_speed;
     }
 
-    /// The ['Instant'] when [`Time::update`] was last called, if it exists
+    /// The ['Instant'] when [`Time::update`] was first called, if it exists.
+    #[inline]
+    pub fn first_update(&self) -> Option<Instant> {
+        self.first_update
+    }
+
+    /// The ['Instant'] when [`Time::update`] was last called, if it exists.
     #[inline]
     pub fn last_update(&self) -> Option<Instant> {
         self.last_update
     }
 
+    /// The [`Instant`] the app was started.
+    #[inline]
+    pub fn startup(&self) -> Instant {
+        self.startup
+    }
+
+    /// The time since startup, as [`Duration`]. Affected by speed multiplier.
+    /// Sampled during the latest update.
+    #[inline]
     pub fn time_since_startup(&self) -> Duration {
+        self.time_since_startup
+    }
+
+    /// The time since startup, as [`f32`] seconds. Affected by speed multiplier.
+    /// Sampled during the latest update.
+    #[inline]
+    pub fn seconds_since_startup(&self) -> f32 {
+        self.time_since_startup().as_secs_f32()
+    }
+
+    /// The time since startup, as [`f64`] seconds. Affected by speed multiplier.
+    /// Sampled during the latest update.
+    #[inline]
+    pub fn seconds_since_startup_f64(&self) -> f64 {
+        self.time_since_startup().as_secs_f64()
+    }
+
+    /// The time since startup, as [`Duration`]. Unaffected by speed multiplier.
+    /// Sampled during the latest update.
+    pub fn real_time_since_startup(&self) -> Duration {
+        self.last_update.unwrap_or(self.startup) - self.startup
+    }
+
+    /// The time since startup, as [`f32`] seconds. Unaffected by speed multiplier.
+    /// Sampled during the latest update.
+    #[inline]
+    pub fn real_seconds_since_startup(&self) -> f32 {
+        self.real_time_since_startup().as_secs_f32()
+    }
+
+    /// The time since startup, as [`f64`] seconds. Unaffected by speed multiplier.
+    /// Sampled during the latest update.
+    #[inline]
+    pub fn real_seconds_since_startup_f64(&self) -> f64 {
+        self.real_time_since_startup().as_secs_f64()
+    }
+
+    /// The time since startup, as [`Duration`]. Unaffected by speed multiplier.
+    /// Sampled at call.
+    pub fn live_real_time_since_startup(&self) -> Duration {
```

Comment:
Now we have "live", "real", "fixed", "variable" times. That's a bit too much. Users can use `Instant::now()` by themselves if they really care about that quantity, no?

Reply:
I was going for parity with Unity.
- `Time.time` => `time.time_since_startup()`
- `Time.unscaledTime` => `time.real_time_since_startup()`
- `Time.realtimeSinceStartup` => `time.live_real_time_since_startup()`

We can bikeshed the naming, but I don't really see the advantage of *not* having all three. I think having them all makes it easier to explain to users which one does what.

Reply:
I didn't know Unity had "realtime". I'm fine leaving it, but I'd avoid "real" and use "CPU time" instead for anything related to `std::time::Instant`. Especially since "real" in Unity terminology is your "live", and your "real" is Unity's "unscaled", which will confused Unity users for sure.

Reply:
While I understand "real" is pretty vague, it isn't `real` vs `live`, it's `real` vs `live_real`. The only difference between the two is that the former is sampled when `Time` updates (at the beginning of the frame) and the latter is sampled wherever you call it.

Reply:
I think I'm on board now with not having the "live" method and just having users do `Instant::now() - time.startup()`. I think it'd only be used for profiling/debugging anyway.

And then have:
- `time.elapsed_since_startup()`
- `time.cpu_elapsed_since_startup()`